build-setup/vscode/using_vscode.md

```diff
@@ -0,0 +1,232 @@
+            "problemMatcher": [
+                "$gcc"
+            ],
+            "group": {
+                "kind": "build",
+                "isDefault": true
+            },
+            "options": {
+                "env": {
+                    "CPP_SRC_HOME": "${env:CPP_SRC_HOME}"
+                }
+            }            
+        },
+        {
+            "label": "Build ml-cpp api tests",
+            "type": "shell",
+            "command": "cd ${workspaceFolder}/lib/api/unittest && make ML_DEBUG=1 -j6",
+            "problemMatcher": [
+                "$gcc"
+            ],
+            "dependsOn": [
+                "Build ml-cpp lib"
+            ],
+            "options": {
+                "env": {
+                    "CPP_SRC_HOME": "${env:CPP_SRC_HOME}"
+                }
+            }              
+        },
+        {
+            "label": "Build ml-cpp core tests",
+            "type": "shell",
+            "command": "cd ${workspaceFolder}/lib/core/unittest && make ML_DEBUG=1 -j6",
+            "problemMatcher": [
+                "$gcc"
+            ],
+            "dependsOn": [
+                "Build ml-cpp lib"
+            ],
+            "options": {
+                "env": {
+                    "CPP_SRC_HOME": "${env:CPP_SRC_HOME}"
+                }
+            }              
+        },
+        {
+            "label": "Build ml-cpp maths tests",
+            "type": "shell",
+            "command": "cd ${workspaceFolder}/lib/maths/unittest && make ML_DEBUG=1 -j6",
+            "problemMatcher": [
+                "$gcc"
+            ],
+            "dependsOn": [
+                "Build ml-cpp lib"
+            ],
+            "options": {
+                "env": {
+                    "CPP_SRC_HOME": "${env:CPP_SRC_HOME}"
+                }
+            }                       
+        },
+        {
+            "label": "Build ml-cpp bin",
+            "type": "shell",
+            "command": "cd ${workspaceFolder}/bin && make ML_DEBUG=1 -j6",
+            "problemMatcher": [
+                "$gcc"
+            ],
+            "dependsOn": [
+                "Build ml-cpp lib"
+            ],
+            "options": {
+                "env": {
+                    "CPP_SRC_HOME": "${env:CPP_SRC_HOME}"
+                }
+            }               
+        }
+    ]
+}
+```
+
+## Debugging
+
+To use [debugging](https://code.visualstudio.com/docs/editor/debugging), you need to specify the binary and the
+arguments in a json file. You can use `preLaunchTask` to specify the user tasks that should be executed before the start
+(e.g. building project).
+
+Here is a example of the `launch.json` file:
+
+```json
+{
+    "version": "0.2.0",
+    "configurations": [
+        {
+            "name": "data_frame_analyzer",
+            "type": "cppdbg",
+            "request": "launch",
+            "program": "${workspaceFolder}/build/distribution/platform/linux-x86_64/bin/data_frame_analyzer",
```

Comment:
> /platform/linux-x86_64/

The platform specific path is the problem here. Perhaps we can define a variable for the value or maybe we just leave a comment saying edit this for your particular platform

Reply:
I added a token for the platform and instructions to set the platform specific value